src/vmm/src/device_manager/persist.rs

```diff
@@ -115,57 +117,59 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                 .lock()
                 .expect("Poisoned lock");
 
-            let mmio_transport = bus_device
-                .as_any()
-                // Only MmioTransport implements BusDevice on x86_64 at this point.
-                .downcast_ref::<MmioTransport>()
-                .expect("Unexpected BusDevice type");
-
-            let transport_state = mmio_transport.save();
-
-            let locked_device = mmio_transport.locked_device();
-            match locked_device.device_type() {
-                TYPE_BLOCK => {
-                    let block_state = locked_device
-                        .as_any()
-                        .downcast_ref::<Block>()
-                        .unwrap()
-                        .save();
-                    states.block_devices.push(ConnectedBlockState {
-                        device_id: device_id.clone(),
-                        device_state: block_state,
-                        transport_state,
-                        mmio_slot: device_info.clone(),
-                    });
-                }
-                TYPE_NET => {
-                    let net_state = locked_device.as_any().downcast_ref::<Net>().unwrap().save();
-                    states.net_devices.push(ConnectedNetState {
-                        device_id: device_id.clone(),
-                        device_state: net_state,
-                        transport_state,
-                        mmio_slot: device_info.clone(),
-                    });
-                }
-                TYPE_VSOCK => {
-                    let vsock = locked_device
-                        .as_any()
-                        // Currently, VsockUnixBackend is the only implementation of VsockBackend.
-                        .downcast_ref::<Vsock<VsockUnixBackend>>()
-                        .unwrap();
-                    let vsock_state = VsockState {
-                        backend: vsock.backend().save(),
-                        frontend: vsock.save(),
-                    };
-                    states.vsock_device = Some(ConnectedVsockState {
-                        device_id: device_id.clone(),
-                        device_state: vsock_state,
-                        transport_state,
-                        mmio_slot: device_info.clone(),
-                    });
-                }
-                _ => unreachable!(),
-            };
+            if let Some(boot_timer) = bus_device.as_any().downcast_ref::<BootTimer>() {
```

Comment:
Why change the code following this `if let` code-block?

I suggest you keep this code block that skips `BootTimer` devices, and leave the rest untouched.

Reply:
Done.